### PR TITLE
Revert "Removes Mergify"

### PR DESCRIPTION
**Pull request by redreceipt**
Reverts NewSpring/ApollosProject#374

### CODE REVIEW

**Comment by redreceipt** [comment]
we can safely put this back. I was worried about `develop` being automatically deleted it, but it can't be via the branch rules.

File: .mergify.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+      - base=develop
+      - label=merge
+      - label!=blocked
+      - title~=$[fix|feat|chore].*
+    actions:
+      merge:
+        method: squash
+        strict: smart
+  - name: delete head branch after merge
+    conditions: []
```
Comment (redreceipt) [inline .mergify.yml]:
```suggestion
    conditions:
      - merged
```

File: .mergify.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+      - title~=$[fix|feat|chore].*
+    actions:
+      merge:
+        method: squash
+        strict: smart
+  - name: delete head branch after merge
+    conditions:
+      - merged
+    actions:
+      delete_head_branch: {}
```
Comment (redreceipt) [inline .mergify.yml:16]:
```suggestion
      delete_head_branch: {}
  - name: automatic merge with strict
    conditions: []
    actions:
      merge:
        method: merge
        strict: true
```

File: .mergify.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  - name: automatic merge with strict
```
Comment (redreceipt) [inline .mergify.yml]:
```suggestion
  - name: automatic update HEAD
```

File: .mergify.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      merge:
+        method: squash
+        strict: smart
+  - name: delete head branch after merge
+    conditions:
+      - merged
+    actions:
+      delete_head_branch: {}
+  - name: automatic update HEAD
+    conditions: []
```
Comment (redreceipt) [inline .mergify.yml]:
```suggestion
    conditions:
      - "#approved-reviews-by>=1"
```

**Comment by richarddubay** [comment]
Which branch rule tells it not to delete `develop`?

**Comment by redreceipt** [comment]
Any branch that has rules can’t be deleted. It says so at the top of that settings page. Plus I noticed last time after merging to Beta, I didn’t even have the option.

—
Michael

**Comment by richarddubay** [comment]
Do we need to give `staging` a rule then so that it won't be deleted once we merge to `master` next week?

**Comment by richarddubay** [comment]
Or I guess it doesn't care about staging because there is no rule. That would actually make more sense.

**Review by richarddubay — APPROVED** [review]

**Pull request merged**